### ISSUE TITLE
Segmentation failt on `time_machine.travel(time_machine.time())`

### ISSUE THREAD

**Issue by GoodWasHere** [adamchainz/time-machine#349]
### Python Version

3.7.15

### Package Version

2.9.0

### Description

Hi, thanks for a great project.

I wanted to create a `traveler` on current time, but I couldn't: 

```
Python 3.7.15 (default, Oct 12 2022, 19:14:38) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import time_machine
>>> time_machine.travel(time_machine.time())
[1]    14791 segmentation fault  /home/user/test/envdev/bin/python
```

It's not fatal, but it's weird.

**Comment by adamchainz** [comment]
`time_machine.time()` is not a documented function so it’s not part of the API. Does the crash also occur if you use `time.time()`?

I would still like to understand and fix the segfault but that isn’t a priority if it only occurs when using the private API.

**Comment by GoodWasHere** [comment]
With `time.time()` everything works fine.

**Comment by adamchainz** [comment]
Thanks for checking. I’ll maybe look into this when I’m next doing a sizeable update to time-machine.

**Comment by adamchainz** [comment]
Fixed in #431 that showed a public API version of the failure.
